checker/check-new-request-non-path-default-parameter.go

```diff
@@ -0,0 +1,40 @@
+package checker
+
+import (
+	"github.com/tufin/oasdiff/diff"
+)
+
+func NewRequestNonPathDefaultParameterCheck(diffReport *diff.Diff, _ *diff.OperationsSourcesMap, config Config) Changes {
+	result := make(Changes, 0)
+	if diffReport.PathsDiff == nil || len(diffReport.PathsDiff.Modified) == 0 {
+		return result
+	}
+	for path, pathItem := range diffReport.PathsDiff.Modified {
+		if pathItem.ParametersDiff == nil || pathItem.Revision == nil || len(pathItem.Revision.Operations()) == 0 {
+			continue
+		}
+		for paramLoc, paramNameList := range pathItem.ParametersDiff.Added {
+			if paramLoc == "path" {
+				continue
+			}
+			for _, param := range pathItem.Revision.Parameters {
+				if !paramNameList.Contains(param.Value.Name) {
+					continue
```

Comment:
I suggest to report this error on each affected endpoint rather than once per path.
The reason is that ApiChange is designed to report a breaking change at the endpoint (operation level), for example, it has an operation field.
Another reason is that the breaking change may only apply to a subset of the operations, for exampple, consider moving a param from op level to path level.

The downside to this approach is that the error may be reported multiple times, one for each operation, but this is ok since we will eventually support the ability to group breaking changes (see https://github.com/Tufin/oasdiff/issues/371).